qemu/qmp.go

```diff
@@ -602,16 +603,30 @@ func (q *QMP) ExecuteQuit(ctx context.Context) error {
 // used to name the device.  As this identifier will be passed directly to QMP,
 // it must obey QMP's naming rules, e,g., it must start with a letter.
 func (q *QMP) ExecuteBlockdevAdd(ctx context.Context, device, blockdevID string) error {
-	args := map[string]interface{}{
-		"options": map[string]interface{}{
+	var args map[string]interface{}
+
+	if q.version.Minor <= 8 {
+		args = map[string]interface{}{
```

Comment:
The only difference between the two maps is the key used for the blockdevID.  It would be better to initialise the map without specifying the blockdevID.  The blockdevID could then be set appropriately in the if statement.  This would save us about 6 lines of code.